docs/guides/c-extension-for-python.mdx

```diff
@@ -0,0 +1,226 @@
+---
+title: Extend Qiskit in Python with C
+description: How to build a C extension for your Qiskit workflow in Python
+---
+
+# Extend Qiskit in Python with C
+
+To accelerate your Qiskit Python programs with C, you can use the Qiskit C extension for Python.
+This requires additional steps to the standalone C usage; for more details, see the [Install the Qiskit C API guide](/guides/install-c-api).
+
+<Admonition type="note">
+  These instructions have only been tested on UNIX-based systems. Windows instructions are in progress._
+</Admonition>
+
+## Requirements
+
+Start by ensuring you have installed the [Qiskit C API](/guides/install-c-api). Next, install the Qiskit Python interface, as follows:
+```bash
+pip install -r requirements.txt -c constraints.txt
+pip install .
+```
+
+## Define the C extension
+
+There are various options to write a C extension for Python. This guide starts with an approach that uses Python's built-in [`ctypes`](https://docs.python.org/3/library/ctypes.html) module, because it requires minimal overhead.  In the next section, the [the Manual C extension](#manual-c-extension) section provides an example of manually building the C extension using Python's C API.
```

Comment:
What do you mean by "overhead" here?  The runtime overhead of `ctypes` is higher than a native C extension.

Reply:
That using `ctypes` requires less boilerplate code, but indeed it's slower so we use the manual extension below. Maybe something like
```suggestion
There are various options to write a C extension for Python. For simplicity, this guide starts with an approach that uses Python's built-in [`ctypes`](https://docs.python.org/3/library/ctypes.html) module.  In the next section, the [the Manual C extension](#manual-c-extension) section provides an example of manually building the C extension using Python's C API for less runtime overhead.
```